l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py

```diff
@@ -53,6 +53,8 @@ def saveAttachment(self, fatturapa, number):
 
         attach_str = fatturapa.to_xml(self.env)
         attach_vals = {
+            "res_model": "fatturapa.attachment.out",
+            "res_id": -1,
```

Comment:
toglierei il  res_id a -1
```suggestion
            
```

Per mettere il res_id corretto  si potrebbe fare una doppia scrittura  create -> update, sostituendo la riga di return con:

```
attach_obj = attach_obj.create(attach_vals)
attach_obj.write({"res_id": attach_obj.id})
return attach_obj
```

Reply:
@jackom83 cosa intendi con "res_id corretto"?

Reply:
Quello dell'allegato che stai creando (l'XML della fattura)

Reply:
Ma questo comporterebbe far comparire l'allegato, come allegato standard, sull'oggetto XML della fattura, giusto? E probabilmente non lo vogliamo.

Comunque vedi https://github.com/OCA/l10n-italy/pull/3827

Reply:
Non mi è chiaro che intendi nn compare gia ? nell'scheda fattura elettronica, forse mi sfugge dove altro comparirebbe ?

Reply:
`res_id` è l'id dell'oggetto a cui l'allegato è... allegato.  Se l'allegato fosse un PDF, res_model sarebbe il modello del documento e res_id il suo ID.

Mettere come res_id l'id dell'allegato stesso creerebbe una sorta di loop, diventerebbe un allegato di sé stesso, non la vedo come una soluzione con più potenziali problemi rispetto a -1,  che non è NULL nel senso che non è un allegato orfano perché è stato cancellato l'oggetto a cui era collegato, ma allo stesso tempo Odoo non lo collega mai a nessuno oggetto visto che hanno tutti ID > 1.

Reply:
@TheMule71 mi convince la tua osservazione, ma allora mi chiedo perche non mettere il res_id dell'oggetto fattura a cui l'allegato si riferisce, è perche altrimenti verrebbe trattato come allegato standard come indicava @eLBati  ? quali controindicazioni ci sono nel consentire cio ? se l'allegato diventa  visibile anche tra gli allegati standard di "account.move" si potrebbe agire sulla vista per nascondere gli allegati di tipo "fatturapa.attachment.out"

Chiaramente è un ipotesi, anche la soluzione in #3827 mi sembra buona, pero vedo che è solo per la 16 se nn erro.

Reply:
Non so perché mi sono perso il commento, comunque un allegato può contenere più fatture.